SearchKit - Allow creation of new records via in-place edit

colemanw opened the pull request:
Overview
----------------------------------------
Allows e.g. an email record to be created if one does not already exist via in-place edit.

Before
----------------------------------------
For joined records (e.g. email), SeachKit in-place edit would not work if one did not already exist.

After
----------------------------------------
Now you can create a new record, same as editing an existing one.

Technical Details
---------------------------
The create params get taken from the join clause. E.g. if you join on an email with a clause like `"location type = work" and "is_primary = true"` then when creating via edit-in-place the new email will have those values (primary, work).

Comments
-----------
@eileenmcnaughton requested this: [dev/core#2853](https://lab.civicrm.org/dev/core/-/issues/2853)

colemanw committed:
SearchKit - Allow creation of new records via in-place edit

Allows e.g. an email record to be created if one does not already exist.
Fixes dev/core#2853

eileenmcnaughton commented:
@colemanw I tested this & was able to create an edit in place email field and could either edit or create using the field. When creating it fill in is_primary and location_type_id were both not null. 

I added other entities (phone, website, related contact & activity) and it did not work for any of them - I can't see any entity hard-coding so it must be in the metadata somewhere?

You can't specify the location type id - I can think of some edge cases where it would be better if you could but out of scope.

colemanw commented:
Ok @eileenmcnaughton, I'll take another look. But first I want to get #22358 resolved because that validation logic will be needed here as well. I'm going to revisit that PR shortly and make any changes needed to resolve your concerns.

colemanw commented:
@eileenmcnaughton I've got this rebased and working now. Added basic validation so you can't enter a blank value when creating a new record, and fixed it to work with multiple joins.

eileenmcnaughton commented:
test this please

eileenmcnaughton commented:
I'm not sure what that fail was (it was one of those ones you have to trawl the console log for) so I've just set it re-running in the hope...

colemanw commented:
@eileenmcnaughton I spotted the problem & fixed. It's passing now.

eileenmcnaughton commented:
OK - seems to work in r-run

allinappli commented:
I confirm it works on 5.47.3 we were able to record an e-mail address when there wasn't one

eileenmcnaughton commented:
thanks for testing @allinappli
